### ISSUE TITLE
Purpose of AfterReady

### ISSUE THREAD

**Issue by filipinyo** [Hyrdaboo/DwarfImpulse#1]
I had a problem if my CameraController was global and the ShakeDirector2D was a child of CameraController. On changing the scene, the ShakeDirector2D would stop working. This function seemed to cause this. Is there any specific reason this exists?

`  
private void AfterReady()
{
    Node currentScene = GetTree().CurrentScene;
    Reparent(currentScene);
    currentScene.MoveChild(this, currentScene.GetChildCount() - 1);
}
`

PS: I really like the plugin! Thank you for sharing it!

**Comment by Hyrdaboo** [comment]
@filipinyo Hi I'm glad you like the plugin. When you said global do you mean autoload? The purpose of AfterReady is to make the ShakeDirector2D the last child of the currently active scene. This is done to ensure the shake is applied after any changes to the transform so that they don't override the shake. The ShakeDirector2D searches for the current scene on start and in your case because your script is on autoload this only happens for the first time and when you change the scene your camera likely overrides the ShakeDirector2D's transform.

TLDR: I haven't taken Autoload into account and this is a bug. Thanks for letting me know I'll fix it asap.

**Comment by Hyrdaboo** [comment]
@filipinyo The bug is fixed now and you can ```new ShakeDirector2D(true)``` now which will make it global. Just make sure your target is also global and if not update it whenever you change or reload scenes. Good luck with your game :)
